### PR TITLE
Task3

### PR DESCRIPTION
**Pull request by michael011299**
Created a get request that returns the categories with testing included

### CODE REVIEW

File: app.js [hunk 1]
```diff
@@ -0,0 +1,17 @@
+const express = require("express");
+const { getCategories } = require("./Controllers/getCategories.controller");
+const app = express();
+app.use(express.json());
+
+app.get("/api/categories", getCategories);
+
+app.all("/*", (req, res) => {
```
Comment (jimstev2447) [inline app.js:8]: you've added a 404 handler but haven't written a test for it, please make sure you are testing this.

File: app.js [hunk 2]
```diff
@@ -0,0 +1,17 @@
+const express = require("express");
+const { getCategories } = require("./Controllers/getCategories.controller");
+const app = express();
+app.use(express.json());
```
Comment (jimstev2447) [inline app.js:4]: Does this need to be here yet? If you aren't parsing any of the bodies of your requests then you needn't add it

File: package.json [hunk 3]
```diff
@@ -23,8 +23,9 @@
   "dependencies": {
     "dotenv": "^16.0.0",
     "express": "^4.18.2",
-    "pg": "^8.7.3",
-    "pg-format": "^1.0.4"
+    "pg": "^8.8.0",
+    "pg-format": "^1.0.4",
+    "supertest": "^6.3.0"
```
Comment (jimstev2447) [inline package.json:28]: should superset be a dependency or a dev dependency?

File: Controllers/getCategories.controller.js [hunk 4]
```diff
@@ -0,0 +1,7 @@
+const { selectCategories } = require("../Models/selectCategories.models");
+
+exports.getCategories = (req, res) => {
+  selectCategories().then((categories) => {
+    res.status(200).send({ categories });
+  });
```
Comment (jimstev2447) [inline Controllers/getCategories.controller.js:6]: dont forget to add a catch block to anywhere you are using promises to ensure your error handling is being used :)

**Review by jimstev2447 — COMMENTED** [review]
Looks good to me, before you merge read through the comments and at least add the 404-bad path test and get a catch block on your selectCategories call. 

Keep up the good work 👍

File: app.js (hunk 1 above)
Comment (michael011299) [reply app.js:8]: Test has been added

**Pull request merged**